data/src/test/java/org/oppia/data/MockTopicService.kt

```diff
@@ -0,0 +1,33 @@
+package org.oppia.data
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import org.oppia.data.backends.gae.NetworkInterceptor
+import org.oppia.data.backends.gae.NetworkSettings
+import org.oppia.data.backends.gae.api.TopicService
+import org.oppia.data.backends.gae.model.GaeTopic
+import retrofit2.Call
+import retrofit2.mock.BehaviorDelegate
+
+/**
+ * Mock TopicService with dummy data from [FakeJsonResponse]
+ */
+class MockTopicService(private val delegate: BehaviorDelegate<TopicService>) : TopicService {
+  override fun getTopicByName(topicName: String): Call<GaeTopic> {
+    val topic = MockGaeTopic()
+    return delegate.returningResponse(topic).getTopicByName(topicName)
+  }
+
+  private fun MockGaeTopic(): GaeTopic {
```

Comment:
Nit: since this is a method, its name should be lowerCamelCase and a verb phrase, e.g. ``createMockGaeTopic``.